### PR TITLE
Add a keybinding to the commit view to easily sign off commits

### PR DESCRIPTION
**Pull request by p3lim**
Partial fix for #36, this will sign-off commits by hitting <kbd>Super</kbd>+<kbd>s</kbd>.

As with my previous PR, this will need proof-checking, I'm still quite new to Python.

### CODE REVIEW

**Comment by TheHippo** [comment]
Why the manual sign off message generation. Couldn't this be done by `git` itself?

**Comment by p3lim** [comment]
I chose this method instead of amending -s (or --signoff) to the commit command (GsCommitViewDoCommitCommand) was to give more of a visual representation to the user that it actually did sign-off.

I suppose my original implementation suggestion from #36 could work fine, but that would require a more verbose output (like suggested in #44) so that the user could verify it.

File: core/commands/commit.py [hunk 1]
```diff
@@ -92,3 +97,26 @@ def run_async(self):
...
+
+        config_name = self.git("config", "user.name").strip()
+        config_email = self.git("config", "user.email").strip()
+
+        sign_text = COMMIT_SIGN_TEXT.format(name=config_name, email=config_email)
+        view_text_list[0] += sign_text
+        view_text_list.insert(1, COMMIT_HELP_TEXT)
+
+        self.view.run_command("gs_replace_view_text", {
+            "text": "".join(view_text_list),
```
Comment (divmain) [inline core/commands/commit.py]: This should read:
```python
"text": COMMIT_HELP_TEXT.join(view_text_list)
```

File: core/commands/commit.py [hunk 2]
```diff
@@ -92,3 +97,26 @@ def run_async(self):
...
+    def run(self, edit):
+        view_text = self.view.substr(sublime.Region(0, self.view.size()))
+        view_text_list = view_text.split(COMMIT_HELP_TEXT)
+
+        config_name = self.git("config", "user.name").strip()
+        config_email = self.git("config", "user.email").strip()
+
+        sign_text = COMMIT_SIGN_TEXT.format(name=config_name, email=config_email)
+        view_text_list[0] += sign_text
+        view_text_list.insert(1, COMMIT_HELP_TEXT)
```
Comment (divmain) [inline core/commands/commit.py]: You can remove this line.

**Comment by divmain** [comment]
Looks good to me.  Can you also update the `COMMIT_HELP_TEXT` to indicate the availability of the shortcut?

**Pull request merged**